test/Constraints/casts.swift

```diff
@@ -180,7 +180,7 @@ var c2f2: C2<[Float]>? = b as! C3
 
 
 // <rdar://problem/15633178>
-var f: (Float) -> Float = { $0 as Float } // expected-warning {{redundant cast to 'Float' has no effect}} {{32-41=}}
+var f: (Float) -> Float = { $0 as Float } 
```

Comment:
I didn't merge my PR because I think this case is supposed to warn, $0 is a float based on contextual type, it can't be anything else...